File: mvp.admin/src/FlexForm.Admin/flexform-admin/src/app/models/dtos/field-definition.model.ts
import { FieldType } from '../enums';
import { ItemSourceDefinition } from './item-source-definition.model';
import { Guid } from 'guid-typescript';
import { Output, EventEmitter } from '@angular/core';
import { BaseProperty } from '../properties';
import { PropertyNameConstants } from '../../property-name-constants';
import { BaseWithTimeStamp } from './base-with-timestamp.model';
import { IClassDefinition } from '../interfaces/class-definition.interface';

export class FieldDefinition extends BaseWithTimeStamp<number> implements IClassDefinition{
    
    name:string;
    entityDefinitionId:number;
    itemSourceDefinitionId:number;
    fieldType:FieldType;
    expression: string;
    itemSourceDefinition:ItemSourceDefinition;
    
    public properties: BaseProperty<any>[];
    color: string;
    public guid: Guid;
    constructor() {
        super();
        this.guid = Guid.create();
    }

}

File: mvp.admin/src/FlexForm.Api.Admin/Model/FieldDefinitionModel.cs
using FlexForm.DDE.Common.Enums;

namespace FlexForm.Api.Admin.Model
{
    public class FieldDefinitionModel : ModelBaseInt
    {
        public string Name { get; set; }

        public int EntityDefinitionId { get; set; }

        public int? ItemSourceDefinitionId { get; set; }

        public FieldType FieldType { get; set; }

        public string Expression { get; set; }

        public ItemSourceDefinitionModel ItemSourceDefinition { get; set; }
    }
}
